pkg/model/fips.go

```diff
@@ -59,6 +61,45 @@ func EnforceGoCompliance(ctx *gotls.Config) {
 		ctx.CipherSuites = fipsGoCiphers
 		ctx.CurvePreferences = []gotls.CurveID{gotls.CurveP256}
 		return
+	case common_features.FIPS_202205:
+		// Mirror the settings from Envoy
+		// https://github.com/envoyproxy/envoy/blob/a0c9c494a4405d78de15c246c0b915d83612f81a/api/envoy/extensions/transport_sockets/tls/v3/common.proto#L49
+		ctx.GetConfigForClient = func(chi *gotls.ClientHelloInfo) (*gotls.Config, error) {
+			// TODO: Go doesn't allow us to change signature requirements so this is all we can do
+			cfg := &gotls.Config{
+				CurvePreferences: []gotls.CurveID{gotls.CurveP256, gotls.CurveP384},
+				MinVersion:       gotls.VersionTLS12,
+			}
+			validSignatureSchemes := map[gotls.SignatureScheme]struct{}{
+				gotls.PKCS1WithSHA256:        {},
+				gotls.PKCS1WithSHA384:        {},
+				gotls.PKCS1WithSHA512:        {},
+				gotls.PSSWithSHA256:          {},
+				gotls.PSSWithSHA384:          {},
+				gotls.PSSWithSHA512:          {},
+				gotls.ECDSAWithP256AndSHA256: {},
+				gotls.ECDSAWithP384AndSHA384: {},
+			}
+			for _, signature := range chi.SignatureSchemes {
```

Comment:
We do this because Go doesn't allow us to set server side signatures. Let's discuss @kyessenov @howardjohn @pmerrison